examples/server/api_like_OAI.py

```diff
@@ -0,0 +1,194 @@
+import argparse
+from flask import Flask, jsonify, request, Response
+import urllib.parse
+import requests
+import time
+import json
+
+
+app = Flask(__name__)
+
+parser = argparse.ArgumentParser(description="An example of using server.cpp with a similar API to OAI. It must be used together with server.cpp.")
+parser.add_argument("--chat-prompt", type=str, help="the top prompt in chat completions(default: A chat between a curious user and an artificial intelligence assistant. The assistant follows the given rules no matter what.)", default='A chat between a curious user and an artificial intelligence assistant. The assistant follows the given rules no matter what.')
+parser.add_argument("--user-name", type=str, help="USER name in chat completions(default: USER:)", default="USER:")
+parser.add_argument("--ai-name", type=str, help="ASSISTANT name in chat completions(default: ASSISTANT:)", default="ASSISTANT:")
+parser.add_argument("--system-name", type=str, help="SYSTEM name in chat completions(default: ASSISTANT's RULE:)", default="ASSISTANT's RULE:")
+parser.add_argument("--stop", type=str, help="the end of response in chat completions(default: </s>)", default="</s>")
+parser.add_argument("--llama-api", type=str, help="Set the address of server.cpp in llama.cpp(default: http://127.0.0.1:8080)", default='http://127.0.0.1:8080')
+parser.add_argument("--api-key", type=str, help="Set the api key to allow only few user(default: NULL)", default="")
+parser.add_argument("--host", type=str, help="Set the ip address to listen.(default: 127.0.0.1)", default='127.0.0.1')
+parser.add_argument("--port", type=int, help="Set the port to listen.(default: 8081)", default=8081)
+
+args = parser.parse_args()
+
+def is_present(json, key):
+    try:
+        buf = json[key]
+    except KeyError:
+        return False
+    return True
+
+#convert chat to prompt
+def convert_chat(messages):
+    prompt = "" + args.chat_prompt + "\n\n"
+
+    for line in messages:
+        if (line["role"] == "system"):
+            prompt += f"{args.system_name} {line['content']}\n"
+        if (line["role"] == "user"):
+            prompt += f"{args.user_name} {line['content']}\n"
+        if (line["role"] == "assistant"):
+            prompt += f"{args.ai_name} {line['content']}{args.stop}\n"
+    prompt += args.ai_name
+
+    return prompt

```

Comment:
I think this could be configurable with some kind of template or something? There are some models that don't even want a `\n` separator.

Reply:
I will include /n in the args.names and args.chat_prompt